persistence/src/commonMain/db_user/com/wire/kalium/persistence/Conversations.sq

```diff
@@ -76,6 +76,88 @@ UPDATE Conversation
+    WHEN "ONE_ON_ONE" THEN  User.user_availability_status
+    WHEN "CONNECTION_PENDING" THEN connection_user.user_availability_status
+    END AS userAvailabilityStatus,
+CASE (Conversation.type)
+    WHEN "ONE_ON_ONE" THEN  User.user_type
+    WHEN "CONNECTION_PENDING" THEN connection_user.user_type
+    END AS userType,
+CASE (Conversation.type)
+    WHEN "ONE_ON_ONE" THEN  User.bot_service
+    WHEN "CONNECTION_PENDING" THEN connection_user.bot_service
+    END AS botService,
+CASE (Conversation.type)
+    WHEN "ONE_ON_ONE" THEN  User.deleted
+    WHEN "CONNECTION_PENDING" THEN connection_user.deleted
+    END AS userDeleted,
+CASE (Conversation.type)
+    WHEN "ONE_ON_ONE" THEN  User.connection_status
+    WHEN "CONNECTION_PENDING" THEN connection_user.connection_status
+    END AS connectionStatus,
+CASE (Conversation.type)
+    WHEN "ONE_ON_ONE" THEN  User.qualified_id
+    WHEN "CONNECTION_PENDING" THEN connection_user.qualified_id
+    END AS otherUserId,
+CASE (Conversation.creator_id)
+    WHEN (SELECT Conversation.qualified_id FROM Conversation WHERE Conversation.type IS "SELF") THEN 1
+    ELSE 0
+    END AS isCreator,
+Conversation.last_notified_message_date AS lastNotifiedMessageDate,
+(SELECT COUNT() FROM Message AS message
+    WHERE message.conversation_id = Conversation.qualified_id
+    AND (DateTime(message.date) >  DateTime(Conversation.last_read_date))
+    AND message.content_type IN ('TEXT', 'ASSET', 'KNOCK', 'MISSED_CALL'))
+    AS unreadMessageCount,
+CASE Conversation.qualified_id
+WHEN (SELECT conversation FROM Member
+        WHERE Member.user =
+        (SELECT Conversation.qualified_id FROM Conversation
+         WHERE Conversation.type IS "SELF")) THEN 0
+    ELSE 1
+    END AS isMember
+FROM Conversation
+LEFT JOIN Member ON Conversation.qualified_id = Member.conversation
+    AND Conversation.type IS "ONE_ON_ONE"
+    AND Member.user IS NOT (SELECT Conversation.qualified_id FROM Conversation WHERE Conversation.type IS "SELF")
+LEFT JOIN User ON User.qualified_id = Member.user AND User.qualified_id
+LEFT JOIN Connection ON Connection.qualified_conversation = Conversation.qualified_id
+    AND (Connection.status = 'SENT'
+         OR Connection.status = 'PENDING'
+         OR Connection.status = "NOT_CONNECTED"
+         AND Conversation.type IS "CONNECTION_PENDING")
+LEFT JOIN User AS connection_user ON Connection.qualified_to = connection_user.qualified_id
+LEFT JOIN Call ON Call.id IS (SELECT MAX(id) FROM Call WHERE Call.conversation_id = Conversation.qualified_id) AND  Call.status IS "STILL_ONGOING";
+
+selectAllConversationsView:
+SELECT * FROM ConversatonsList
+WHERE type IS NOT "SELF" AND
+(type IS "GROUP" AND (name IS NOT NULL OR otherUserId IS NOT NULL) --filter deleted groups after first sync
+OR (type IS NOT "GROUP" AND otherUserId IS NOT NULL)) -- show other conversation- todo: problem here! if the sync wasn't succesful the the user seems to be null!
+ORDER BY lastModifiedDate DESC;
+
```

Comment:
holy shit I did not know this is possible, great job

Reply:
It's beautiful, isn't it?

I've seen big boys working server-side, but this is the first time I see such big queries wile working on mobile apps.

"We are not making a JSON Reader" could be our team's motto.

Reply:
![](https://static.wikia.nocookie.net/meme/images/f/f9/Mother-of-god-meme_3997206.jpg/revision/latest?cb=20140417165010)